fix: 81134: display conflicted taint without a json representation

manugupt1 opened the pull request:
From
https://github.com/kubernetes/kubernetes/issues/81134#issuecomment-540800845

1. string is formatted as unsafe json. Since this is created on the fly;
TaintKey and Effect are formatted as a 'key=effect' instead of '{key=effect}'
eliminating unsafe json representation.

2. Adds a fail test case for the case where a taint with same key and effect is added
and removed together.

Manual Testing

```
▶ ./_output/bin/kubectl taint nodes kind-control-plane key1=:NoSchedule key1=:NoSchedule-
error: can not both modify and remove the following taint(s) in the same command: key1=NoSchedule
```

Also, ran unit tests 

```
▶ make test WHAT=./staging/src/k8s.io/kubectl/pkg/cmd/taint
+++ [0729 16:32:27] Running tests without code coverage and with -race
ok  	k8s.io/kubernetes/staging/src/k8s.io/kubectl/pkg/cmd/taint	0.278s
```

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind bug

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
Earlier the string that was constructed was an unsafe json. 
As recommended in https://github.com/kubernetes/kubernetes/issues/81134 I am adding an in-depth check
to not create a json as araw string.

#### Which issue(s) this PR fixes:
https://github.com/kubernetes/kubernetes/issues/81134

#### Special notes for your reviewer:
Please see Manual Testing to see that the user output did not change.

#### Does this PR introduce a user-facing change?

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
1. Changes json representation for a conflicted taint to Key=Effect when a conflicted taint occurs in kubectl taint.
```

eddiezane commented on staging/src/k8s.io/kubectl/pkg/cmd/taint/taint.go:
```diff
@@ -241,20 +241,29 @@ func (o TaintOptions) Validate() error {
 				continue
 			}
 			if len(taintRemove.Effect) == 0 || taintAdd.Effect == taintRemove.Effect {
-				conflictTaint := fmt.Sprintf("{\"%s\":\"%s\"}", taintRemove.Key, taintRemove.Effect)
```
Let's scrap the json altogether here and just make these `key=value`.

eddiezane replied:
We don't even output the conflicts for labels 😅

https://github.com/kubernetes/kubernetes/blob/fc85deff7faef7f349742bba599696175ba920d4/staging/src/k8s.io/kubectl/pkg/cmd/label/label.go#L431

manugupt1 replied:
Thank you for reviewing the PR. I have made the requested changes.

rikatz replied:
@eddiezane before issuing the lgtm here, I think that your point in the second review is: we should also print the "conflictTaints" right?

@manugupt1 maybe doing something like:

return nil, nil, fmt.Errorf("can not both modify and remove a label in the same command: %v", conflictTaint) 

or creating a pretifier just for this array would be really helpful for users :D

eddiezane replied:
@rikatz sorry for the confusion. The return error is happening already below

https://github.com/kubernetes/kubernetes/blob/5f35d48104cf285511adffeed8bee79207ee00cc/staging/src/k8s.io/kubectl/pkg/cmd/taint/taint.go#L257

rikatz replied:
Thanks Eddie, I didn't went to check above :)

eddiezane reviewed (changes requested):
Thanks for the PR!

eddiezane commented:
/triage accepted
/priority backlog
/ok-to-test

Commit:
fix: 81134: display conflicted taint without a json representation.

1. string is formatted as unsafe json. Since this is created on the fly;
TaintKey and Effect are formatted as a 'key=effect' instead of '{key=effect}'
eliminating unsafe json representation.

2. Adds a fail test case for the case where a taint with same key and effect is added
and removed together.

Manual Testing

```
▶ ./_output/bin/kubectl taint nodes kind-control-plane key1=:NoSchedule key1=:NoSchedule-
error: can not both modify and remove the following taint(s) in the same command: key1=NoSchedule
```

eddiezane commented:
/approve

rikatz commented:
/lgtm
Thanks!

manugupt1 commented:
/test pull-kubernetes-integration

Failure looks like it is related to calico.

manugupt1 commented:
@rikatz can you please give another LGTM as I had rebased my changes earlier. Thanks!

rikatz commented:
/lgtm
Thanks!

dims commented:
/ok-to-test

pacoxu commented:
/retest
